src/boot/efi/efi-string.c

```diff
@@ -37,14 +37,6 @@
 DEFINE_STRNLEN(char, strnlen8);
 DEFINE_STRNLEN(char16_t, strnlen16);
 
-size_t strlen8(const char *s) {
```

Comment:
I just realized I can provide inline forms for these too (similar to strcmp) making this commit unnecessary…:
```c
#define DEFINE_STRNLEN(type, suffix)                                                     \
        static inline size_t _strnlen##suffix(const type *s, bool until_end, size_t n) { \
                if (!s)                                                                  \
                        return 0;                                                        \
                                                                                         \
                size_t len = 0;                                                          \
                while ((until_end || len < n) && *s) {                                   \
                        s++;                                                             \
                        len++;                                                           \
                }                                                                        \
                                                                                         \
                return len;                                                              \
        }                                                                                \
        size_t strnlen##suffix(const type *s, size_t n) {                                \
                return _strnlen##suffix(s, false, n);                                    \
        }                                                                                \
        size_t strlen##suffix(const type *s) {                                           \
                return _strnlen##suffix(s, true, 0);                                     \
        }
```

Reply:
Then again, I am not sure if shaving off a compare like that gains anything…

Reply:
Yeah, I do not think it provides anything better... But if you'd like, please do it.